src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        MachineProducer producer = type == Truck.class ? new TruckProducer()
+                : type == Bulldozer.class ? new BulldozerProducer()
+                : type == Excavator.class ? new ExcavatorProducer() : null;
+        return producer != null ? producer.get() : new ArrayList();
+    }
```

Comment:
I think that here is better to use if else, because ternar operation is too difficult to read in this case